GOVERNANCE.md

```diff
@@ -0,0 +1,136 @@
+Governance for the R data.table project
+
+# Purpose and scope
+
+## This document
+
+The purpose of this document is to define how people related to the project work together, so that the project can expand to handle a larger and more diverse group of contributors.
+
+## The R package
+
+The purpose of the project is to maintain the R data.table package, which is guided by the following principles:
+
+* No external Imports/LinkingTo/Depends dependencies (external meaning those not maintained by the project)
+* Few (if any) Suggests/Enhances dependencies
+* Time & memory efficiency
+* Concise syntax (minimal redundancy in code)
+* Stable code base (strong preference for user-friendly back-compatibility with data.table itself and with old versions of R)
+* Comprehensive and accessible documentation and run-time signals (errors, warnings)
+
+To prioritize developer time, we define what is in and out of current scope. Feature requests in issues and pull requests that are out of current scope should be closed immediately, because they are not the current priority. If someone wants to contribute code that is currently out of scope, they first have to make a pull request that changes the scope as defined below.
```

Comment:
This seems a bit contrary to me regarding the current `Contributing` wiki entry, in particular 
> So, before you spend the time getting to that stage, it may save you time to create an issue first and start a discussion to see if your idea would be accepted in principle. If you are going to spend more than a day on the PR, creating an issue first lets other people know you are working on it to save duplicating effort.

In any case we should check to keep consistency between wiki and the Governance.md

Reply:
What seems contrary? What would you suggest changing? 
It seems to me that the wiki is consistent. It is a good idea to create an issue first to make sure the idea would be accepted in principle (=is in scope).

Reply:
But being currently out of scope does not mean that it would not be accepted if somebody is willing to put the time into it (particularly thinking about large file or database support). That's what I'm finding contradictory.

Reply:
Creating issue first before submitting PR is always good choice. Any design discussions (+ in//out scope) should be taking place in the issue, and implementation discussions in the PR. I know that reality is/will be very different, but at least that should be our aim.